DuckDuckGo/SecureVaultErrorReporter.swift

```diff
@@ -30,8 +30,10 @@ final class SecureVaultErrorReporter: SecureVaultErrorReporting {
         guard !ProcessInfo().arguments.contains("testing") else { return }
 #endif
         switch error {
-        case .initFailed, .failedToOpenDatabase:
-            Pixel.fire(pixel: .secureVaultInitError)
+        case .initFailed(let error):
+            Pixel.fire(pixel: .secureVaultInitFailedError, error: error)
+        case .failedToOpenDatabase(let error):
+            Pixel.fire(pixel: .secureVaultFailedToOpenDatabaseError, error: error)
```

Comment:
One issue here: the `Pixel.fire(pixel:)` and `Pixel.fire(pixel:, error:)` calls have different default behaviour when it comes to the additional parameters they send. The ones that include an error don't include ATB by default, but the version without the error _does_ send it by default.

I don't think we should include ATB with these pixels, unless you have a counterargument for how it would be used. If not, let's change the call below this comment to leave out the additional parameters.

Reply:
That's a good point, I didn't notice that, will change

Reply:
Tbh, we should probably change what this does by default and by default not send extra info